source/expectations/matchers/ut_equal.tpb

```diff
@@ -186,13 +198,32 @@ create or replace type body ut_equal as
     return l_result;
   end;
 
+  member procedure exclude(self in ut_equal, a_items varchar2) is
+  begin
+    exclude( ut_varchar2_list( a_items ) );
+  end;
+
+  member procedure exclude(self in ut_equal, a_items ut_varchar2_list) is
+    l_result ut_equal := self;
+  begin
+    l_result.options.exclude.add_items(a_items);
+    l_result.expectation.to_(l_result );
```

Comment:
all calls to: `l_result.expecatation` in this type are assuming that the `expectation` attribute is set.

Reply:
Given the number of places this is called, it would be good to delegate exception handling to a helper procedure.

Reply:
@lwasylow 
I don't really remember why this change was done.
I think it's part of some clean-up operation but not really part of new matcher. Is your memory any better than mine?

Reply:
@jgebal 

Went back to old conversations on slack from may 27th 2020 and found this:

_To not duplicate old methods from matchers we can change behaviour.
UT_MATCHER is additional attribute of expectation which is set when function is called.
Procedure TO_EQUAL in UT_EXPECTATION works the same._

Reply:
Yeah. I've found it. But after 2 years I can't even understand what I had in mind.
I'll put the original (polish) version here, just in case, as the free version of SLACK that we use purges messages history.
>Propozycja:
Żeby nie duplikować wszystkich metod z matcherów w ut_Expectation  można zmienić to zachowanie w następujący sposób:
ut_matcher a teraz dodatkowy atrybut expectation który jest ustawiany gdy wołana jest funcja.
Procedura to_equal w ut_Expectation  działa jak dotychczas - raportuje expecattaion
Funkcja to_equal  w ut_Expecattion_compound :
ustawia wartość pola expectation w przekazanym matcher  na wartość self
zwraca matcher, tak żeby metchod chaining teraz już używał metod matchera (dzięki temu nie trzeba już duplikować dodatkowych metod w ut_expectation
Matcher może odwołać się teraz do expectation z którego został wywołany.
W procedurze matchera (ostatnia metoda w łańcuchu) self.expecation is null  matcher nie woła expecataion, w innym wypadku procedury  matcher wołaję metodę to_ przekazując self 
W ten sposób składnia:
ut.expect(...).to_( equal(...).incude(...) ) sprawia, że:
matcher nie woła expectation, bo expectation nie jest ustawione w matcher
matcher jest przekazany do expectation i expectation wykonuje sprawdzenie wartości i raportuje
Składnia:
ut.expect(...).to_equal(...).include sprawia że:
to_equal ustawia self jako expectation w matcherze, zwraca matcher i dokonuje się metchod chaining
include  wywoływane jest na matcherze i poniewarz jest to procedura i pole expectation jest ustawione, matcher woła metodę  ecpectation.to_  przekazując self jako arcument
Powinno działać, ale jest 2 w nocy więc mogę majaczyć